File: src/main/java/com/flexpoker/table/command/repository/InMemoryTableEventRepository.java
package com.flexpoker.table.command.repository;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.UUID;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Repository;

import com.flexpoker.exception.FlexPokerException;
import com.flexpoker.table.command.framework.TableEvent;

@Profile("default")
@Repository
public class InMemoryTableEventRepository implements TableEventRepository {

    private final Map<UUID, List<TableEvent>> tableEventMap;

    public InMemoryTableEventRepository() {
        tableEventMap = new HashMap<>();
    }

    @Override
    public List<TableEvent> fetchAll(UUID id) {
        return tableEventMap.get(id);
    }

    @Override
    public List<TableEvent> setEventVersionsAndSave(int basedOnVersion, List<TableEvent> events) {
        var aggregateId = events.get(0).getAggregateId();

        if (!tableEventMap.containsKey(aggregateId)) {
            tableEventMap.put(aggregateId, new ArrayList<>());
        }

        var existingEvents = tableEventMap.get(aggregateId);
        if (existingEvents.size() != basedOnVersion) {
            throw new FlexPokerException("events to save are based on a different version of the aggregate");
        }

        for (int i = 0; i < events.size(); i++) {
            events.get(i).setVersion(basedOnVersion + i + 1);
        }

        tableEventMap.get(aggregateId).addAll(events);

        return events;
    }

}

File: src/main/kotlin/com/flexpoker/table/command/repository/InMemoryTableEventRepository.kt
package com.flexpoker.table.command.repository

import com.flexpoker.config.ProfileNames
import com.flexpoker.exception.FlexPokerException
import com.flexpoker.table.command.events.TableEvent
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Repository
import java.util.ArrayList
import java.util.HashMap
import java.util.UUID

@Profile(ProfileNames.DEFAULT, ProfileNames.TABLE_COMMAND_INMEMORY)
@Repository
class InMemoryTableEventRepository : TableEventRepository {

    private val tableEventMap: MutableMap<UUID, MutableList<TableEvent>> = HashMap()

    override fun fetchAll(id: UUID): List<TableEvent> {
        return tableEventMap[id]!!
    }

    override fun setEventVersionsAndSave(basedOnVersion: Int, events: List<TableEvent>): List<TableEvent> {
        val aggregateId = events[0].aggregateId
        if (!tableEventMap.containsKey(aggregateId)) {
            tableEventMap[aggregateId] = ArrayList()
        }
        val existingEvents: List<TableEvent> = tableEventMap[aggregateId]!!
        if (existingEvents.size != basedOnVersion) {
            throw FlexPokerException("events to save are based on a different version of the aggregate")
        }
        for (i in events.indices) {
            events[i].version = basedOnVersion + i + 1
        }
        tableEventMap[aggregateId]!!.addAll(events)
        return events
    }

}